VISION.md

```diff
@@ -0,0 +1,178 @@
+# Vision
+
+This document describes the vision for this crate. It is a work-in-progress
+document and may be adapted over the development time of the codebase.
+
+## config-rs
+
+The effort to write this crate is made to revolutionize and evolve the
+[config-rs] crate and solve some design issues with it as well as doing a
+complete rework of its interface and API to streamline it and clean it up.
+
+### Features
+
+The [config-rs] crate is a crate that helps developers _read_ configuration. It
+does not (yet?) offer functionality to _alter_ configuration more than
+in-process, so there's no functionality for writing an adapted configuration
+back to disc.
+
+The crate offers a _layering_ technique, which can be used to "merge"
+configuration files as "layers". For example, a system-wide configuration file
```

Comment:
imo an important aspect of layering is that any error for a config field can be pointed back to the layer source it came from